packages/flutter/lib/src/widgets/routes.dart

```diff
@@ -558,21 +558,103 @@ abstract class ModalRoute<T> extends TransitionRoute<T> with LocalHistoryRoute<T
+  ///       body: new Center(
+  ///         child: new Text('Hello World'),
+  ///       ),
+  ///     );
+  ///   },
+  ///   transitionsBuilder: (
+  ///       BuildContext context,
+  ///       Animation<double> primaryAnimation,
+  ///       Animation<double> secondaryAnimation,
+  ///       Widget child) {
+  ///     return new SlideTransition(
+  ///       position: new FractionalOffsetTween(
+  ///         begin: FractionalOffset.bottomLeft,
+  ///         end: FractionalOffset.topLeft
+  ///       ).animate(primaryAnimation),
+  ///       child: child, // child is the value returned by pageBuilder
+  ///     );
+  ///   },
+  /// );
+  ///```
+  ///
+  /// We've used PageRouteBuilder to demonstrate the buildTransitions method
+  /// here. The body of an override of the buildTransitions method would be
+  /// defined in the same way.
+  ///
+  /// When the Navigator pushes a route on the top of its stack, the
+  /// secondaryAnimation can be used to define how route that was on the top
+  /// of the stack leaves the screen. Similarly when the topmost route is
+  /// popped, the secondaryAnimation can be used to define how the route
+  /// below it reappears on the screen. When the Navigator pushes a new route
+  /// on the top of its stack, the old topmost route's secondaryAnimation
+  /// runs from 1.0 to 0.0.  When the Navigator pops the topmost route, the
+  /// secondaryAnimation for the route below it runs from 0.0 to 1.0.
+  ///
+  /// The example below adds an transition that's driven by the
+  /// secondaryAnimation. When this route disappears because a new route has
+  /// been pushed on top of it, it translates in the opposite direction of
+  /// the new route. Likewise when the route is exposed because the topmost
+  /// route has been popped off.
+  ///
+  /// ```dart
+  ///   transitionsBuilder: (
+  ///       BuildContext context,
+  ///       Animation<double> primaryAnimation,
+  ///       Animation<double> secondaryAnimation,
+  ///       Widget child) {
+  ///     return new SlideTransition(
+  ///       position: new FractionalOffsetTween(
+  ///         begin: FractionalOffset.bottomLeft,
+  ///         end: FractionalOffset.topLeft,
+  ///       ).animate(primaryAnimation),
+  ///       child: new SlideTransition(
+  ///         position: new FractionalOffsetTween(
+  ///           begin: FractionalOffset.topLeft,
+  ///           end: FractionalOffset.bottomLeft,
+  ///         ).animate(secondaryAnimation),
+  ///         child: child,
+  ///       ),
+  ///     );
+  ///   }
+  ///```
+  ///
+  /// In practice the secondaryAnimation is not used very often because animating
+  /// both the topmost route and the route below it can be more distracting
+  /// than helpful.
+  ///
   /// * [context] The context in which the route is being built.
-  /// * [animation] The animation for this route's transition. When entering,
+  /// * [primaryAnimation] The animation for this route's transition. When entering,
```

Comment:
I think it's important to note that this parameter is rarely used but I've left out my opnion as to why.